exercises/practice/hamming/hamming_test.go

```diff
@@ -4,39 +4,30 @@ import "testing"
 
 func TestHamming(t *testing.T) {
 	for _, tc := range testCases {
-		got, err := Distance(tc.s1, tc.s2)
-		if tc.expectError {
-			// check if err is of error type
-			var _ error = err
-
-			// we expect error
-			if err == nil {
-				t.Fatalf("Distance(%q, %q); expected error, got nil.",
-					tc.s1, tc.s2)
-			}
-		} else {
-			// we do not expect error
-			if err != nil {
-				t.Fatalf("Distance(%q, %q) returned unexpected error: %v",
-					tc.s1, tc.s2, err)
-			}
-			if got != tc.want {
-				t.Fatalf("Distance(%q, %q) = %d, want %d.",
-					tc.s1, tc.s2, got, tc.want)
+		t.Run(tc.description, func(t *testing.T) {
+			got, err := Distance(tc.s1, tc.s2)
+			if tc.expectError {
+				if err == nil {
```

Comment:
nested if can be avoided by using `&&`

Reply:
With `if tc.expectError && err == nil` we have one problem: If an error is exepected (`tc.expectError` is `true`) and err is not nil, we would go to the `else`, but I don't think we want to, do we?

Reply:
yes, you are right, it would only work if you would add `!tc.expectError` to the conditions below

there is another problem now, the linter does not like the `else if`, instead of the suggested switch maybe just if statements (no else) with Fatalf inside would be better

sorry, I didn't know trying to improve this would lead to such a rabbit hole, feel free to just go back to the old nested version and we leave it at that :see_no_evil:

Reply:
I have replaced it with the `switch` as I have also used this in other exercises and I like it :)